standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java

```diff
@@ -49,6 +50,8 @@ public DataSource create(Configuration hdpConfig, int maxPoolSize) throws SQLExc
     String driverUrl = DataSourceProvider.getMetastoreJdbcDriverUrl(hdpConfig);
     String user = DataSourceProvider.getMetastoreJdbcUser(hdpConfig);
     String passwd = DataSourceProvider.getMetastoreJdbcPasswd(hdpConfig);
+    long maxLifeTime = MetastoreConf.getLongVar(hdpConfig,
+            MetastoreConf.ConfVars.METASTORE_HIKARICP_MAX_LIFE_TIME);
```

Comment:
Do we need a seprate config, there is a logic below
```
    Properties properties = replacePrefix(DataSourceProvider.getPrefixedProperties(hdpConfig, HIKARI));

    HikariConfig config;
    try {
      config = new HikariConfig(properties);
```

I believe that should take care of setting maxLifetime

Reply:
No, we don't, need to have it, probably. Please see the discussion in this PR and the test case I added.
https://github.com/apache/hive/pull/5639
https://github.com/apache/hive/pull/5639/files#diff-21013298e701863201669c0a08dce28bff302bca654dda0287b76cb344ae2c7a